### PR TITLE
Adding recipe for pyahocorasick.

### PR DESCRIPTION
**Pull request by hellothisisMatt**
This is a new recipe for Pyahocorasick.
`https://github.com/WojciechMula/pyahocorasick`

### CODE REVIEW

File: recipes/pyahocorasick/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  sha256: {{ sha256sum }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
```
Comment (synapticarbors) [inline recipes/pyahocorasick/meta.yaml]: please add `toolchain` since this compiles C-extensions.

File: recipes/pyahocorasick/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  home: https://github.com/WojciechMula/pyahocorasick
+  license: BSD 3-clause
+  license_family: BSD
+  license_file: LICENSE
+  summary: 'Python module (C extension and plain python) implementing Aho-Corasick algorithm.'
+  description: |
+      pyahocorasick is a fast and memory efficient library for exact or approximate multi-pattern string 
+      search meaning that you can find multiple key strings occurrences at once in some input text. 
+      The library provides an ahocorasick Python module that you can use as a plain dict-like Trie or 
+      convert a Trie to an automaton for efficient Aho-Corasick search.
```
Comment (synapticarbors) [inline recipes/pyahocorasick/meta.yaml]:
```
dev_url: https://github.com/WojciechMula/pyahocorasick
doc_url: http://pyahocorasick.readthedocs.io/en/latest/
```

Comment (hellothisisMatt) [reply recipes/pyahocorasick/meta.yaml]: Added :)

File: recipes/pyahocorasick/meta.yaml (hunk 1 above)
Comment (hellothisisMatt) [reply recipes/pyahocorasick/meta.yaml]: Added in 9b6887a

**Pull request merged**